clojure-ts-mode.el

```diff
@@ -1691,10 +1697,191 @@ Forms between BEG and END are aligned according to
+  (let ((threading-sexp (clojure-ts--threading-sexp-node)))
+    (save-excursion
+      (clojure-ts--skip-first-child threading-sexp)
+      (raise-sexp))))
+
+(defun clojure-ts--fix-sexp-whitespace ()
+  "Fix whitespace after unwinding a threading form."
+  (save-excursion
+    (let ((beg (point)))
+      (treesit-end-of-thing 'sexp)
+      (indent-region beg (point))
+      (delete-trailing-whitespace beg (point)))))
+
+(defun clojure-ts--unwind-sexps-counter ()
+  "Return total number of s-expressions of a threading form at point."
+  (if-let* ((threading-sexp (clojure-ts--threading-sexp-node)))
+      (save-excursion
+        (clojure-ts--skip-first-child threading-sexp)
+        (let ((n 0))
+          (while (treesit-end-of-thing 'sexp 1 'restricted)
+            (setq n (1+ n)))
+          n))
+    (user-error "Nothing to unwind at point")))
+
+(defun clojure-ts-unwind (&optional n)
+  "Unwind thread at point or above point by N levels.
+
+With universal argument \\[universal-argument], fully unwinds thread."
+  (interactive "P")
+  (setq n (cond
+           ((equal n '(4)) (clojure-ts--unwind-sexps-counter))
+           (n)
+           (1)))
+  (if-let* ((threading-sexp (clojure-ts--threading-sexp-node))
+            (sym (thread-first threading-sexp
+                               (treesit-node-child 0 t)
+                               (clojure-ts--named-node-text))))
+      (save-excursion
+        (let ((beg (thread-first threading-sexp
+                                 (treesit-node-start)
+                                 (copy-marker)))
+              (end (thread-first threading-sexp
+                                 (treesit-node-end)
+                                 (copy-marker))))
+          (while (> n 0)
+            (cond
+             ((string-match-p (rx bol (* "some") "->" eol) sym)
+              (clojure-ts--unwind-first))
+             ((string-match-p (rx bol (* "some") "->>" eol) sym)
+              (clojure-ts--unwind-last)))
+            (setq n (1- n))
+            ;; After unwinding we check if it is the last expression and maybe
+            ;; splice it.
+            (when (clojure-ts--nothing-more-to-unwind)
+              (clojure-ts--pop-out-of-threading)
+              (clojure-ts--fix-sexp-whitespace)
+              (setq n 0)))
+          (indent-region beg end)
+          (delete-trailing-whitespace beg end)))
+    (user-error "Nothing to unwind at point")))
+
+(defun clojure-ts-unwind-all ()
+  "Fully unwind thread at point or above point."
+  (interactive)
+  (clojure-ts-unwind '(4)))
+
+(defvar clojure-ts-refactor-map
+  (let ((map (make-sparse-keymap)))
+    (keymap-set map "C-u" #'clojure-ts-unwind)
+    (keymap-set map "u" #'clojure-ts-unwind)
+    map))
+
 (defvar clojure-ts-mode-map
   (let ((map (make-sparse-keymap)))
     ;;(set-keymap-parent map clojure-mode-map)
     (keymap-set map "C-c SPC" #'clojure-ts-align)
+    (keymap-set map clojure-ts-refactor-map-prefix clojure-ts-refactor-map)
```

Comment:
Now that we have some useful commands in the mode, it's a good idea to add a mode menu as well, e.g. by using `easymenu` (see how it's done in `clojure-mode`).